frontend/simple/js/events.js

```diff
@@ -3,13 +3,26 @@ import * as Events from '../../../shared/events'
 import store from './state'
 import backend from '../js/backend'
 import {namespace} from '../js/backend/hapi'
+import _ from 'lodash'
 
 export class GroupContract extends Events.HashableGroup {
   static vuex = GroupContract.Vuex({
-    state: { votes: [], payments: [], members: [], invitees: [], profiles: {} },
+    state: { proposals: {}, payments: [], members: [], invitees: [], profiles: {} },
     mutations: {
       HashableGroupPayment (state, {data}) { state.payments.push(data) },
-      HashableGroupVote (state, {data}) { state.votes.push(data) },
+      HashableGroupProposal (state, {data, hash}) { state.proposals[hash] = {...data, for: [], against: []} },
+      HashableGroupVoteForProposal (state, {data}) {
+        if (state.proposals[data.proposalHash]) {
+          state.proposals[data.proposalHash].for.push(data.username)
+          if (state.proposals[data.proposalHash].for.length >= state.proposals[data.proposalHash].threshold) { Vue.delete(state.proposals, data.proposalHash) }
```

Comment:
Something else does happen from the vote.vue; it will perform the action. I can modify the threshold